dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -465,148 +517,154 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                         }
                     }
                 }
-            } else if value.ty(&self.body.local_decls, self.tcx).is_unsafe_ptr() {
-                if let Rvalue::Use(p) = value {
-                    location.statement_index += 1;
-                    if p.place().map(|p| p.is_indirect()).unwrap_or(false) {
-                        // We're dereferencing a pointer, the result of which is another pointer;
-                        // trace the destination
-                        let mut loc = location;
-                        loc.statement_index += 1;
-                        self.add_instrumentation_point(
-                            location,
-                            load_value_fn,
-                            vec![InstrumentationOperand::RawPtr(Operand::Copy(dest))],
-                            false,
-                            false,
-                            EventMetadata {
-                                source: None,
-                                destination: Some(to_mir_place(&dest)),
-                                transfer_kind: TransferKind::None,
-                            },
-                        );
-                    } else {
-                        // Copying a pointer to another place; trace the destination
-                        self.add_instrumentation_point(
-                            location,
-                            copy_fn,
-                            vec![InstrumentationOperand::RawPtr(Operand::Copy(dest))],
-                            false,
-                            false,
-                            EventMetadata {
-                                source: p.place().as_ref().map(to_mir_place),
-                                destination: Some(to_mir_place(&dest)),
-                                transfer_kind: TransferKind::None,
-                            },
-                        );
+            } else if is_region_or_unsafe_ptr(value_ty) {
+                match value {
+                    Rvalue::Use(Operand::Copy(p)) | Rvalue::Use(Operand::Move(p)) => {
+                        location.statement_index += 1;
+                        if p.is_indirect() {
+                            // We're dereferencing something, the result of which is another pointer;
+                            // trace the destination
+                            self.add_instrumentation_point(
```

Comment:
You could just set `source` and `instrument_fn` locals in both sides of this if and combine the instrumentation call, saving some redundant code.

Reply:
will do in #552